homeassistant/components/hue/v2/light.py

```diff
@@ -204,6 +205,9 @@ async def async_turn_on(self, **kwargs: Any) -> None:
         xy_color = kwargs.get(ATTR_XY_COLOR)
         color_temp = normalize_hue_colortemp(kwargs.get(ATTR_COLOR_TEMP))
         brightness = normalize_hue_brightness(kwargs.get(ATTR_BRIGHTNESS))
+        if self._restore_brightness and brightness is None:
```

Comment:
This is a workaround so we need to add a comment here why this workaround is here including a reference to a bug report at the Hue developers forum. So please add a comment that there is a bug in the Hue bridge that setting a transition at turn off (re)sets the brightness to 1 at the next turn on cycle.

Reply:
This is a list of form post already reporting this bug to Philips Hue, but appereantly they are not so keen on fixing it:
https://developers.meethue.com/forum/t/brightness-turns-down-to-1-automatically-shortly-after-sending-off-signal-hue-bug/5692
https://developers.meethue.com/forum/t/lights-turn-on-with-lowest-brightness-via-siri-if-turned-off-via-api/6700
https://developers.meethue.com/forum/t/using-transitiontime-with-on-false-resets-bri-to-1/4585
https://developers.meethue.com/forum/t/bri-value-changing-in-switching-lights-on-off/6323
https://developers.meethue.com/forum/t/fade-in-fade-out/6673

Apperently it comes from the lack of a zigbee command to turn off the bulb with a transition, instead the bridge sends a zigbee command to dim to 1% and then sends a second command to turn off. that causes the bridge to save the brightness as 1% causing this issue.

However they can simply fix this by caching the brightness before sending the transition to 1% command, simular as to what I do in this PR (but then in the bridge firmware).

Reply:
Yeah, I agree, but my comment still applies, please link to (one of the) upstream bug report in our in-line comment so we can track this in the longer term and others reading the code (even in a few years) can still know why this workaround exist in our code to fix the Hue bug.

Reply:
Yea, was already doing that, but you were faster.
comment now pushed.